### PR TITLE
Refactor cluster registry to use Secret to store kubeconf

### PR DESCRIPTION
**Pull request by sbezverk**
4th attempt to resolve Circle CI stuck on Waiting for status to be reported issue.

This PR refactors cluster registry to use k8s Secret object to store kubeconf files of member clusters.

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: clusterregistry
  namespace: istio-system
data:
  kube-3.yaml: |-
    apiVersion: clusterregistry.k8s.io/v1alpha1
    kind: Cluster
    metadata:
      name: "kube-3"
      annotations:
        config.istio.io/pilotEndpoint: "192.168.80.234:9080"
        config.istio.io/platform: "Kubernetes"
        config.istio.io/pilotCfgStore: True
        config.istio.io/accessConfigSecret: "kube-3.conf"   < ---   Name of Secret which stores kube-3 kubeconf config 
        config.istio.io/accessConfigSecretNamespace: "istio-system"   < ---   Namespace where secrets will be stored. 
```

### CODE REVIEW

File: pilot/pkg/config/clusterregistry/clusterregistry.gotmpl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: clusterregistry.k8s.io/v1alpha1
+kind: {{.Kind}}
+metadata:
+  name: {{.Name}}
+  annotations:
+    config.istio.io/pilotEndpoint: "{{.PilotIP}}:9080"
```
Comment (rshriram) [inline pilot/pkg/config/clusterregistry/clusterregistry.gotmpl]: make the port configurable as well. 9080 is for old v1 stuff.

Comment (sbezverk) [reply pilot/pkg/config/clusterregistry/clusterregistry.gotmpl]: this file is ONLY used in unit testing, so no any exposure to any inside or outside service.

**Pull request merged**